packages/three-vrm-springbone/src/VRMSpringBoneManager.ts

```diff
@@ -81,134 +98,122 @@ export class VRMSpringBoneManager {
+      springBone.update(delta);
 
       // update children world matrices
       // it is required when the spring bone chain is sparse
-      traverseChildrenUntilConditionMet(springBone.bone, (object) => {
+      traverseChildrenUntilConditionMet(springBone.bone, this._relevantChildrenUpdated);
+    }
+  }
+
+  /**
+   * Sorts the joints ensuring they are updated in the correct order taking dependencies into account.
+   */
+  private _sortJoints() {
+    const springBoneOrder: Array<VRMSpringBoneJoint> = [];
+    const springBonesTried = new Set<VRMSpringBoneJoint>();
+    const springBonesDone = new Set<VRMSpringBoneJoint>();
+    const ancestors = new Set<THREE.Object3D>();
+    for (const springBone of this._joints) {
+      this._insertJointSort(springBone, springBonesTried, springBonesDone, springBoneOrder, ancestors);
+    }
+    this._sortedJoints = springBoneOrder;
+
+    const lca = lowestCommonAncestor(ancestors);
+    this._ancestors = [];
+    if (lca) {
+      this._ancestors.push(lca);
+      traverseChildrenUntilConditionMet(lca, (object: THREE.Object3D) => {
         // if the object has attached springbone, halt the traversal
         if ((this._objectSpringBonesMap.get(object)?.size ?? 0) > 0) {
           return true;
         }
-
-        // otherwise update its world matrix
-        object.updateWorldMatrix(false, false);
+        this._ancestors.push(object);
         return false;
       });
     }
   }
 
-  /**
-   * Update a spring bone.
-   * If there are other spring bone that are dependant, it will try to update them recursively.
-   * It updates matrixWorld of all ancestors and myself.
-   * It might log an warning message if there are any circular dependencies.
-   *
-   * Intended to be used in {@link update} and {@link _processSpringBone} itself recursively.
-   *
-   * @param springBone A springBone you want to update
-   * @param springBonesTried Set of springBones that are already tried to be updated
-   * @param springBonesDone Set of springBones that are already up to date
-   * @param objectUpdated Set of object3D whose matrixWorld is updated
-   */
-  private _processSpringBone(
+  private _insertJointSort(
     springBone: VRMSpringBoneJoint,
     springBonesTried: Set<VRMSpringBoneJoint>,
     springBonesDone: Set<VRMSpringBoneJoint>,
-    objectUpdated: Set<THREE.Object3D>,
-    callback: (springBone: VRMSpringBoneJoint) => void,
-  ): void {
+    springBoneOrder: Array<VRMSpringBoneJoint>,
+    ancestors: Set<THREE.Object3D>,
+  ) {
     if (springBonesDone.has(springBone)) {
       return;
     }
 
     if (springBonesTried.has(springBone)) {
       if (!this._hasWarnedCircularDependency) {
-        console.warn('VRMSpringBoneManager: Circular dependency detected while updating springbones');
+        console.warn('VRMSpringBoneManager: Circular dependency detected');
         this._hasWarnedCircularDependency = true;
       }
       return;
     }
-    springBonesTried.add(springBone);
```

Comment:
@mrxz We forgot to put this line to `_insertJointSort`!

I want to ask you why you deleted this line, just in case. Have you encountered any issues with having this line? If you simply forgot, it's all fine.

Reply:
@0b5vr I must have accidentality removed that line while moving code around, sorry. It should be added back in.

Reply:
I see I see, thank you for the explanation!

Reply:
A PR adds the line: https://github.com/pixiv/three-vrm/pull/1556#issue-2735402313